### PR TITLE
Remove hard-coded log file before and after installation to prevent segmentation fault (#44)

### LINKED ISSUE #44: Segmentation fault on non-root installation
- [x] I read [the conda-forge documentation](https://conda-forge.org/docs/user/introduction.html#how-can-i-install-packages-from-conda-forge) and could not find the solution for my problem there.

Issue: `cudatoolkit-dev` throws segmentation fault then aborts, on non-root installation at post-link step. Full error log is as followings:

```
Collecting package metadata (current_repodata.json): done
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/jnooree/anaconda3/envs/test

  added / updated specs:
    - cudatoolkit-dev=11.1
    - python=3.9


The following NEW packages will be INSTALLED:

  _libgcc_mutex      conda-forge/linux-64::_libgcc_mutex-0.1-conda_forge
  _openmp_mutex      conda-forge/linux-64::_openmp_mutex-4.5-1_gnu
  ca-certificates    conda-forge/linux-64::ca-certificates-2021.5.30-ha878542_0
  cudatoolkit-dev    conda-forge/linux-64::cudatoolkit-dev-11.1.1-py39h3811e60_0
  ld_impl_linux-64   conda-forge/linux-64::ld_impl_linux-64-2.36.1-hea4e1c9_2
  libffi             conda-forge/linux-64::libffi-3.3-h58526e2_2
  libgcc-ng          conda-forge/linux-64::libgcc-ng-11.1.0-hc902ee8_8
  libgomp            conda-forge/linux-64::libgomp-11.1.0-hc902ee8_8
  libstdcxx-ng       conda-forge/linux-64::libstdcxx-ng-11.1.0-h56837e0_8
  ncurses            conda-forge/linux-64::ncurses-6.2-h58526e2_4
  openssl            conda-forge/linux-64::openssl-1.1.1l-h7f98852_0
  pip                conda-forge/noarch::pip-21.2.4-pyhd8ed1ab_0
  python             conda-forge/linux-64::python-3.9.7-h49503c6_0_cpython
  python_abi         conda-forge/linux-64::python_abi-3.9-2_cp39
  readline           conda-forge/linux-64::readline-8.1-h46c0cb4_0
  setuptools         conda-forge/linux-64::setuptools-58.0.3-py39hf3d152e_0
  sqlite             conda-forge/linux-64::sqlite-3.36.0-h9cd32fc_0
  tk                 conda-forge/linux-64::tk-8.6.11-h27826a3_1
  tzdata             conda-forge/noarch::tzdata-2021a-he74cb21_1
  wheel              conda-forge/noarch::wheel-0.37.0-pyhd8ed1ab_1
  xz                 conda-forge/linux-64::xz-5.2.5-h516909a_1
  zlib               conda-forge/linux-64::zlib-1.2.11-h516909a_1010


Proceed ([y]/n)?

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
ERROR conda.core.link:_execute(699): An error occurred while installing package 'conda-forge::cudatoolkit-dev-11.1.1-py39h3811e60_0'.
Rolling back transaction: done

LinkError: post-link script failed for package conda-forge::cudatoolkit-dev-11.1.1-py39h3811e60_0
location of failed script: /home/jnooree/anaconda3/envs/test/bin/.cudatoolkit-dev-post-link.sh
==> script messages <==
<None>
==> script output <==
stdout: Log file not open.
Running Post installation
downloading https://developer.download.nvidia.com/compute/cuda/11.1.1/local_installers/cuda_11.1.1_455.32.00_linux.run to /home/jnooree/anaconda3/envs/fgpred/pkgs/cudatoolkit-dev/cuda_11.1.1_455.32.00_linux.run
Extracting on Linux

stderr: Segmentation fault (core dumped)
Traceback (most recent call last):
  File "/home/jnooree/anaconda3/envs/test/bin/cudatoolkit-dev-post-install.py", line 216, in <module>
    _main()
  File "/home/jnooree/anaconda3/envs/test/bin/cudatoolkit-dev-post-install.py", line 212, in _main
    extractor.extract()
  File "/home/jnooree/anaconda3/envs/test/bin/cudatoolkit-dev-post-install.py", line 124, in extract
    subprocess.run(cmd, env=os.environ.copy(), check=True)
  File "/home/jnooree/anaconda3/envs/test/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/home/jnooree/anaconda3/envs/test/pkgs/cudatoolkit-dev/cuda_11.1.1_455.32.00_linux.run', '--silent', '--toolkit', '--toolkitpath=/tmp/tmp09e_7h5l', '--override']' returned non-zero exit status 139.

return code: 1

()
```

In my other experiment, it succeed with administrator privileges.

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list
```
Nothing. It was fresh installed via `conda create -c conda-forge -n test python=3.9 cudatoolkit-dev=11.1`
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info

     active environment : None
            shell level : 0
       user config file : /home/jnooree/.condarc
 populated config files : /home/jnooree/.condarc
          conda version : 4.10.3
    conda-build version : not installed
         python version : 3.9.7.final.0
       virtual packages : __linux=4.4.0=0
                          __glibc=2.23=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /home/jnooree/anaconda3  (writable)
      conda av data dir : /home/jnooree/anaconda3/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/jnooree/anaconda3/pkgs
                          /home/jnooree/.conda/pkgs
       envs directories : /home/jnooree/anaconda3/envs
                          /home/jnooree/.conda/envs
               platform : linux-64
             user-agent : conda/4.10.3 requests/2.26.0 CPython/3.9.7 Linux/4.4.0-201-generic ubuntu/16.04.7 glibc/2.23
                UID:GID : 1027:500
             netrc file : /home/jnooree/.netrc
           offline mode : False

```
</details>

### PR DESCRIPTION
**Pull request by jnooree**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Resolves conda-forge/cudatoolkit-dev-feedstock#44.

### CODE REVIEW

**Comment by jnooree** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {% set name = "cudatoolkit-dev" %}
 {% set release = "11.4" %}
-{% set version_build = "0" %}
+{% set version_build = "1" %}
```
Comment (nnkwame) [inline recipe/meta.yaml]: Hi jnooree - to bump the build number, you want to adjust the value at line 17 instead.

Else, If the intention is to build for a different version of cudatoolkit, then you will need to adjust additionally the driver versions for both platforms (lines 4 and 5). This way the correct files to retrieve can be resolved by the cudatoolkit-dev installer.

The right version numbers can be derived from the naming format of the runfile / executable. See [this](https://developer.nvidia.com/cuda-11-4-1-download-archive?target_os=Linux&target_arch=x86_64&Distribution=Ubuntu&target_version=20.04&target_type=runfile_local) for linux cudatoolkit 11.4.1 as example.

Comment (jnooree) [reply recipe/meta.yaml]: Resolved by 32d9713. Thank you very much!

File: recipe/cudatoolkit-dev-post-install.py [hunk 2]
```diff
@@ -176,11 +187,11 @@ def extract(self):
                     'Something went wrong in executing `{}`: directory `{}` does not exist'
                     .format(' '.join(cmd), toolkitpath))
 
-			# Install files directly to the library prefix. 
-			# This is because Windows 10 requires either admin privileges or developer mode enabled (since Creators Update) for the creation of symlinks.
-			# These options are not guaranteed at the user end
+# Install files directly to the library prefix.
```
Comment (AgrawalAmey) [inline recipe/cudatoolkit-dev-post-install.py]: Nit: fix indent

File: .azure-pipelines/azure-pipelines-win.yml [hunk 3]
```diff
@@ -95,7 +95,7 @@ jobs:
 
     - script: |
         call activate base
-        conda.exe build "recipe" -m .ci_support\%CONFIG%.yaml
+        conda.exe build "recipe" -m .ci_support\%CONFIG%.yaml --suppress-variables
```
Comment (AgrawalAmey) [inline .azure-pipelines/azure-pipelines-win.yml:98]: What does this option do?

**Review by AgrawalAmey — APPROVED** [review]

**Review by nnkwame — APPROVED** [review]

File: recipe/cudatoolkit-dev-post-install.py (hunk 2 above)
Comment (jnooree) [reply recipe/cudatoolkit-dev-post-install.py]: Fixed by 621a835.

**Pull request merged**